### PR TITLE
Fix:vLLM Model Supported check throwing circular dependency

### PR DESCRIPTION
**Pull request by gavrissh**
**What this PR does / why we need it**:
After vLLM update to version 4.2
The load of Llama Arch was causing a circular dependency error.

This PR fixes that issue

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests you ran to verify your changes and the relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [x] Ran sanity checks on Llama, Mistral, and Gemma arch Models

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.

### CODE REVIEW

File: python/huggingfaceserver/huggingfaceserver/vllm/utils.py [hunk 1]
```diff
@@ -37,15 +36,15 @@ def vllm_available() -> bool:
 
 def infer_vllm_supported_from_model_architecture(
     model_config_path: Union[Path, str],
-) -> Optional[Type[nn.Module]]:
+) -> bool:
     if not _vllm:
-        return None
+        return False
     model_config = AutoConfig.from_pretrained(model_config_path)
     architecture = model_config.architectures[0]
```
Comment (spolti) [inline python/huggingfaceserver/huggingfaceserver/vllm/utils.py]: don't we need to check all archs here?

File: python/huggingfaceserver/huggingfaceserver/vllm/utils.py [hunk 2]
```diff
@@ -37,15 +36,15 @@ def vllm_available() -> bool:
...
-    model_cls = ModelRegistry.load_model_cls(architecture)
-    if model_cls is None:
+
+    if architecture not in ModelRegistry.get_supported_archs():
         logger.info("not a supported model by vLLM")
```
Comment (spolti) [inline python/huggingfaceserver/huggingfaceserver/vllm/utils.py]: shouldn't it return false if there is no supported archs?

Comment (gavrissh) [reply python/huggingfaceserver/huggingfaceserver/vllm/utils.py]: missed it, Thanks

File: python/huggingfaceserver/huggingfaceserver/vllm/utils.py (hunk 1 above)
Comment (gavrissh) [reply python/huggingfaceserver/huggingfaceserver/vllm/utils.py]: We generally noticed a case of a single arch in response. 
Updated, with changes to check all.

**Pull request merged**